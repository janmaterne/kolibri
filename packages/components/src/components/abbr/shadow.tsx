import type { JSX } from '@stencil/core';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import type { AbbrAPI, AbbrStates, LabelPropType } from '../../schema';
import { validateLabel } from '../../schema';

/**
 * @slot - The abbreviation (short form).
 */
@Component({
	tag: 'kol-abbr',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAbbr implements AbbrAPI {
	public render(): JSX.Element {
		return (
			<Host class="kol-abbr">
				<abbr>
					<slot />
				</abbr>
				{this.state._label ? ` (${this.state._label})` : ''}
			</Host>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label?: LabelPropType;

	@State() public state: AbbrStates = {
		_label: '', // âš  required
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label);
	}
}
