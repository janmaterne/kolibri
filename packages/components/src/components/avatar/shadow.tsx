import type { JSX } from '@stencil/core';
import { Component, h, Prop } from '@stencil/core';

import type { AvatarProps } from '../../schema';
import { KolAvatarWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-avatar',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAvatar implements AvatarProps {
	public render(): JSX.Element {
		return <KolAvatarWcTag _src={this._src} _label={this._label}></KolAvatarWcTag>;
	}

	/**
	 * Sets the image `src` attribute to the given string.
	 */
	@Prop() public _src?: string;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: string;
}
