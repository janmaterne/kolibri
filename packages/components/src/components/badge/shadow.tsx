import type { BadgeAPI, BadgeStates, ButtonProps, KoliBriIconsProp, LabelPropType, PropColor, Stringified } from '../../schema';
import { featureHint, handleColorChange, objectObjectHandler, parseJson, setState, validateColor, validateIcons } from '../../schema';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import { KolSpanFc } from '@fc';

import { nonce } from '../../utils/dev.utils';

import type { JSX } from '@stencil/core';
import { KolButtonWcTag } from '../../core/component-names';
featureHint(`[KolBadge] Optimierung des _color-Properties (rgba, rgb, hex usw.).`);

@Component({
	tag: 'kol-badge',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolBadge implements BadgeAPI {
	private bgColorStr = '#000';
	private colorStr = '#fff';
	private readonly id = nonce();

	private renderSmartButton(props: ButtonProps): JSX.Element {
		return (
			<KolButtonWcTag
				_ariaControls={this.id}
				_customClass={props._customClass}
				_disabled={props._disabled}
				_hideLabel={true}
				_icons={props._icons}
				_id={props._id}
				_label={props._label}
				_on={props._on}
				_tooltipAlign={props._tooltipAlign}
				_variant={props._variant}
			></KolButtonWcTag>
		);
	}

	public render(): JSX.Element {
		const hasSmartButton = typeof this.state._smartButton === 'object' && this.state._smartButton !== null;

		return (
			<Host class="kol-badge">
				<span
					class={{
						'smart-button': typeof this.state._smartButton === 'object' && this.state._smartButton !== null,
					}}
					style={{
						backgroundColor: this.bgColorStr,
						color: this.colorStr,
					}}
				>
					<KolSpanFc id={hasSmartButton ? this.id : undefined} allowMarkdown icons={this.state._icons} label={this._label} />
					{hasSmartButton && this.renderSmartButton(this.state._smartButton as ButtonProps)}
				</span>
			</Host>
		);
	}

	/**
	 * Defines the backgroundColor and foregroundColor.
	 */
	@Prop() public _color?: Stringified<PropColor> = '#000';

	/**
	 * Defines the icon classnames (e.g. `_icons="fa-solid fa-user"`).
	 */
	@Prop() public _icons?: Stringified<KoliBriIconsProp>;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Allows to add a button with an arbitrary action within the element (_hide-label only).
	 */
	@Prop() public _smartButton?: Stringified<ButtonProps>;

	@State() public state: BadgeStates = {
		_color: {
			backgroundColor: '#000',
			foregroundColor: '#fff',
		},
		_icons: {},
	};

	private handleColorChange = (value: unknown) => {
		const colorPair = handleColorChange(value);
		this.bgColorStr = colorPair.backgroundColor;
		this.colorStr = colorPair.foregroundColor as string;
	};

	@Watch('_icons')
	public validateIcons(value?: KoliBriIconsProp): void {
		validateIcons(this, value);
	}

	@Watch('_color')
	public validateColor(value?: Stringified<PropColor>): void {
		validateColor(this, value, {
			defaultValue: '#000',
			hooks: {
				beforePatch: this.handleColorChange,
			},
		});
	}

	@Watch('_smartButton')
	public validateSmartButton(value?: ButtonProps | string): void {
		objectObjectHandler(value, () => {
			try {
				value = parseJson<ButtonProps>(value as string);
				// eslint-disable-next-line no-empty
			} catch (e) {
				// value behält den ursprünglichen Wert
			}
			setState(this, '_smartButton', value);
		});
	}

	public componentWillLoad(): void {
		this.validateIcons(this._icons);
		this.validateColor(this._color);
		this.validateSmartButton(this._smartButton);
	}
}
