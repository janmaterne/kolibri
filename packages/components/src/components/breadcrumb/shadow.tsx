import type { BreadcrumbAPI, BreadcrumbLinkProps, BreadcrumbStates, LabelPropType, LinkProps, Stringified } from '../../schema';
import { a11yHintLabelingLandmarks, validateLabel } from '../../schema';
import { Component, Fragment, h, Prop, State, Watch } from '@stencil/core';

import { addNavLabel, removeNavLabel } from '../../utils/unique-nav-labels';
import { watchNavLinks } from '../nav/validation';

import type { JSX } from '@stencil/core';
import { KolIconTag, KolLinkTag } from '../../core/component-names';

@Component({
	tag: 'kol-breadcrumb',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolBreadcrumb implements BreadcrumbAPI {
	private readonly renderLink = (link: BreadcrumbLinkProps, index: number): JSX.Element => {
		const lastIndex = this.state._links.length - 1;
		return (
			<li class="kol-breadcrumb__list-element" key={index}>
				{index !== 0 && <KolIconTag class="kol-breadcrumb__icon" _label="" _icons="codicon codicon-chevron-right" />}
				{index === lastIndex ? (
					<span class="kol-breadcrumb__list-element-span">
						{link._hideLabel ? (
							<KolIconTag
								class="kol-breadcrumb__icon"
								_label={link._label}
								_icons={typeof link._icons === 'string' ? link._icons : 'codicon codicon-symbol-event'}
							/>
						) : (
							<>{link._label}</>
						)}
					</span>
				) : (
					<KolLinkTag class="kol-breadcrumb__link" {...link}></KolLinkTag>
				)}
			</li>
		);
	};

	public render(): JSX.Element {
		return (
			<nav class="kol-breadcrumb" aria-label={this.state._label}>
				<ul class="kol-breadcrumb__list">
					{this.state._links.length === 0 && (
						<li>
							<KolIconTag class="kol-breadcrumb_icon" _label="" _icons="codicon codicon-home" />…
						</li>
					)}
					{this.state._links.map(this.renderLink)}
				</ul>
			</nav>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Defines the list of links combined with their labels to render.
	 */
	@Prop() public _links!: Stringified<BreadcrumbLinkProps[]>;

	@State() public state: BreadcrumbStates = {
		_label: '', // ⚠ required
		_links: [],
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType, _oldValue?: LabelPropType, initial = false): void {
		if (!initial) {
			removeNavLabel(this.state._label); // remove the current
		}
		validateLabel(this, value, {
			required: true,
		});
		a11yHintLabelingLandmarks(value);
		addNavLabel(this.state._label); // add the state instead of prop, because the prop could be invalid and not set as new label
	}

	@Watch('_links')
	public validateLinks(value?: Stringified<LinkProps[]>): void {
		watchNavLinks('KolBreadcrumb', this, value);
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label, undefined, true);
		this.validateLinks(this._links);
	}

	public disconnectedCallback(): void {
		removeNavLabel(this.state._label);
	}
}
