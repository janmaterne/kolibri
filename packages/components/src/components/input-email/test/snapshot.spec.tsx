import { KolInputEmailTag } from '@component-names';
import type { InputEmailProps } from '@schema';
import { executeInputSnapshotTests } from '@testing';

import { KolInputEmail } from '../shadow';

executeInputSnapshotTests<InputEmailProps>(
	KolInputEmailTag,
	[KolInputEmail],
	{
		_value: 'email@test.de',
	},
	{ hasSmartButton: true },
);

executeInputSnapshotTests<InputEmailProps>(
	KolInputEmailTag,
	[KolInputEmail],
	{
		_value: 'email@test.de',
		_suggestions: ['email1@test.de', 'email2@test.de', 'email3@test.de'],
	},
	{ hasSmartButton: true },
);
