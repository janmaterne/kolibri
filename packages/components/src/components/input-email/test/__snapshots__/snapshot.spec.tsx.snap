// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _accessKey="V" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _alert=true 1`] = `
<kol-input-email _alert="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _alert="" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _disabled=true 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _hint="Hint" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-email _hide-error="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _readOnly=true 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _readonly="" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _shortKey="S" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _shortkey="S" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          S
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _accessKey="V" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _accesskey="V" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          V
        </span>
      </span>
      <div slot="input">
        <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _alert=true 1`] = `
<kol-input-email _alert="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _alert="" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _disabled=true 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _disabled="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _hint="Hint" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="Hint" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-email _hide-error="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hideerror="" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _readOnly=true 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _readonly="" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _shortKey="S" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _shortkey="S" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        Label
        <span aria-hidden="true" class="access-key-hint">
          S
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _touched=true 1`] = `
<kol-input-email _touched="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _alert="" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _touched=true 1`] = `
<kol-input-email _touched="" class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _alert="" _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" _touched="" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" 1`] = `
<kol-input-email class="has-value kol-input-email">
  <template shadowrootmode="open">
    <kol-input _currentlength="17" _hint="" _id="id-nonce" _label="Label" _tooltipalign="top" class="email" role="presentation">
      <span slot="label">
        <span>
          Label
        </span>
      </span>
      <div slot="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" spellcheck="false" type="email" value="email@example.com">
      </div>
    </kol-input>
  </template>
</kol-input-email>
`;
